doc/content/getting_started/contributing.md

```diff
@@ -0,0 +1,244 @@
+   [have SSH set up on your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account))):
+
+   ```
+   cd ~/projects
+   git clone git@github.com:<your_user_name>/tmap8.git
+   ```
+
+   In your local copy, this clone URL is given the shorthand name "origin". This will become relevant
+   later in this document.
+
+### Add an upstream git remote
+
+!style halign=left
+Add the main TMAP8 repository as an "upstream" git remote, so that you can
+[git fetch](https://git-scm.org/docs/git-fetch) updates from the main repository:
+
+```
+cd ~/projects/tmap8
+git remote add upstream git@github.com:idaholab/tmap8.git
+```
+
+To fetch changes, run:
+
+```
+git fetch upstream
+```
+
+## Creating a new branch
+
+Before make any changes locally (which would require having a working version of the application),
+we will need to create a new branch.
+
+1. In your terminal, go to your application:
+
+   ```
+   cd ~/projects/tmap8
+   ```
+
+1. It is usually best to make sure your changes are based on devel:
+
+   ```
+   git checkout devel
+   ```
+
+1. Clean and reset devel to get it up to date (make sure you do not have any changes on devel, because you would lose that):
+
+   ```
+   git reset --hard upstream/devel && git submodule update --init
+   ```
+
+1. Create new branch:
+
+   ```
+   git checkout -b <BRANCH_NAME>
+   ```
+
+## Make modifications and commit (the regular development workflow)
+
+At this point, you have a local branch that is up to date with the upstream remote. You’re now ready
+to make your changes. Remember, **small, consistent changes** are much better than dumping massive,
+unrelated changes all at once.
+
+1. Make your changes locally to address the issue (partly or fully) - using
+   [VSCode is recommended](help/development/VSCode.md).
+1. In your terminal, go to your application:
+
+   ```
+   cd ~/projects/tmap8
+   ```
+
+1. Make sure that the app still compiles and that the tests run:
+
+   ```
+   make -j6 && ./run_tests -j6
+   ```
+
+   If you modified the documentation, you can then build and test it locally:
+
+   ```
+   cd ~/projects/tmap8/doc && ./moosedocs.py build --serve
+   ```
+
+1. Type `git status` to see the status of your branch. That should show you the changed files and
+   give you some commands to stage them.
+1. Once you are ready, perform
+
+   ```
+   git add <FILE_NAME_1> <FILE_NAME_2> <FILE_NAME_3> ...
+   ```
+
+   to stage your updated files.
+1. At that point, it is usually good practice to run
+
+   ```
+   git clang-format HEAD~1
+   ```
+
+   to fix the formatting of the files based on the `.clang-format` file in the root of the repository.
+   If files are changed, run
+
+   ```
+   git add <...>
+   ```
+
+   for these files.
+1. Now you’re ready to commit. Run
+
+   ```
+   git commit
+   ```
+
+1. A file editor will appear. Press the `i` key, and you’ll be able to type a short description of
```

Comment:
These instructions are only relevant to those who have VIM / VI as their default Terminal editor. This is not a requirement of UNIX-compliant terminals. The language should be adjusted here clarifying that we are assuming that VI / VIM is the editor being used (which is *usually* the default for Linux and Mac.....Ubuntu notably used to use Nano - and might still! - so we must be careful).

Reply:
I updated the language to incorporate your comment.